From 06d4dee43f17a92f3814f96cf8a05795e6a04425 Mon Sep 17 00:00:00 2001
From: Ionut Vicovan <Ionut.Vicovan@nxp.com>
Date: Fri, 22 May 2020 09:38:23 +0000
Subject: [PATCH] env: Add Makefile rule to generate default environment

Add new rule `env-default` to dump the default
environment from env/built-in.o to a file named
`env-default.txt`

Signed-off-by: Vicovan Ionut-Valentin <Ionut.Vicovan@nxp.com>
---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index d1aa6f7..7a3c658 100644
--- a/Makefile
+++ b/Makefile
@@ -810,6 +810,7 @@ endif
 
 # Always append ALL so that arch config.mk's can add custom ones
 ALL-y += u-boot.srec u-boot.bin u-boot.sym System.map binary_size_check
+ALL-y += env-default
 
 ALL-$(CONFIG_ONENAND_U_BOOT) += u-boot-onenand.bin
 ifeq ($(CONFIG_SPL_FSL_PBL),y)
@@ -1046,6 +1047,10 @@ u-boot.ldr:	u-boot
 		$(LDR) -T $(CONFIG_CPU) -c $@ $< $(LDR_FLAGS)
 		$(BOARD_SIZE_CHECK)
 
+# Extract default environment to a file
+env-default: env/built-in.o FORCE
+	$(shell $(CONFIG_SHELL) $(srctree)/scripts/get_default_envs.sh $(build-dir) > $@.txt)
+
 # binman
 # ---------------------------------------------------------------------------
 quiet_cmd_binman = BINMAN  $@
-- 
2.7.4

